src/ImportCsvDialog.cpp

```diff
@@ -412,6 +449,55 @@ void ImportCsvDialog::importCsv(const QString& fileName)
     }
 }
 
+sqlb::FieldVector ImportCsvDialog::fetchCsvHeader(const QString& fileName)
+{
+    // Parse all csv data
+    QFile file(fileName);
+    file.open(QIODevice::ReadOnly);
+
+    CSVParser csv(ui->checkBoxTrimFields->isChecked(), currentSeparatorChar(), currentQuoteChar());
+    csv.setCSVProgress(new CSVImportProgress(file.size()));
+
+    QTextStream tstream(&file);
+    tstream.setCodec(currentEncoding().toUtf8());
+    csv.parse(tstream);
```

Comment:
When fetching just the headers, you should be able to pass ```1``` as the second argument to ```parse()```, like this:
```c++
csv.parse(tstream, 1);
```

This limits the number of records to just the first one which should be enough for the header. This way you don't have to parse the entire file and (if this code isn't shared with ```importCsv()```) you can get rid of the progress dialog entirely to simplify the code a bit and prevent any flickering from quickly opening and closing windows.